### ISSUE TITLE
Why does the project configuration editor buffer?

### ISSUE THREAD

**Issue by nleroy917** [pepkit/pephub#105]
When you go to the edit page of any PEP, the project configuration `yaml` editor (monaco editor) buffers for two seconds.

The issue is that there is a hard-coded `setTimeout()` call to load the editor while we wait for JavaScript and assets to be pulled from the CDN.

**Comment by nleroy917** [comment]
I believe I have identified the `<script />` that was taking time to load and causing this. I added an `id="monaco"` to the tag and then added an event listener:

```js
const script = document.getElementById("monaco")
  script.addEventListener("load", () => {
    editor = loadYAMLEditor()
    editor.onKeyUp(detectYamlChanges)
    getProjectConfigYamlFromDatabase()
  })
```

**Comment by nsheff** [comment]
instead of a hard-coded setTimeout, can't we just `await` the loading ?

Or is that basically what you've done with this eventListener, so no more setTimeout is required?

**Comment by nleroy917** [comment]
I think that's what this is. This is all in the HTML, so I don't think `await` means anything in this sense?

**Comment by nleroy917** [comment]
It was discovered that this **does not** work in chrome and we need to find another solution

**Comment by nleroy917** [comment]
This will be solved if we release a new version with a React-based SPA

**Comment by nleroy917** [comment]
This is now **definitely** solved in the latest release (`0.7.x`) with the new React SPA 😎
